UnitTest/AlgorithmsTests/BinarySearcherTest.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using Xunit;
+using Algorithms.Search;
+
+namespace UnitTest.AlgorithmsTests
+{
+    public static class BinarySearcherTest
```

Comment:
Please add a test case for the enumerator functionality, i.e.: create a new binary searcher on a list of unsorted integers, and enumerate over them, asserting a correct order.